tests/test_nodeset.py

```diff
@@ -66,3 +67,54 @@ def test_ranges_overlap(ranges1, ranges2, expected):
 def test_ranges_vec_overlap(ranges1, vec, expected):
     out = _ranges_vec_overlap(ranges1, vec)
     numpy.testing.assert_array_equal(out, expected)
+
+
+@pytest.fixture
+def nodeset_files(tmpdir):
+    """
+    Generates example nodeset files
+    """
+    # Define nodesets
+    config_nodeset_file = {
+        "Mosaic": {
+            "population": "default"
+        },
+        "Layer1": {
+            "layer": 1
+        },
+    }
+
+    simulation_nodesets_file = {
+        "Mosaic": {
+            "population": "All"
+        },
+        "Layer2": {
+            "layer": 2
+        },
+    }
+
+    # Create the test JSON files with nodesets
+    config_nodeset = tmpdir.join("node_sets_circuit.json")
+    with open(config_nodeset, 'w') as f:
+        json.dump(config_nodeset_file, f)
+
+    simulation_nodeset = tmpdir.join("node_sets_simulation.json")
+    with open(simulation_nodeset, 'w') as f:
+        json.dump(simulation_nodesets_file, f)
+
+    yield config_nodeset, simulation_nodeset
```

Comment:
We could delete the files after they'r used, that's why generally we used `yield`s